be/src/exprs/lambda_function.cpp

```diff
@@ -28,6 +28,10 @@ namespace starrocks {
 LambdaFunction::LambdaFunction(const TExprNode& node) : Expr(node, false), _common_sub_expr_num(node.output_column) {}
 
 Status LambdaFunction::prepare(starrocks::RuntimeState* state, starrocks::ExprContext* context) {
+    if (_is_prepared) {
+        return Status::OK();
+    }
+    _is_prepared = true;
```

Comment:
This should be set to false in the close function.

Reply:
@imay the rf push down expr to scan, then a expr will be shared among scan and project. The project will prepare the expr, following the scan also prepare the expr, then project-open, scan-open .... scan-close, project-close.

Reply:
what is the relation between open, prepare and close?

Reply:
prepare->open->eval->close is a whole calling chain. prepare may init some states, which should be exist until deleting the expr. open-eval-close is called per task, like in some scan tasks.